sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -1358,13 +1401,20 @@ abstract class CastBase extends UnaryExpression with TimeZoneAwareExpression wit
     case StringType =>
       val wrapper = ctx.freshVariable("intWrapper", classOf[UTF8String.IntWrapper])
       (c, evPrim, evNull) =>
+        val casting = if (ansiEnabled) {
+          s"$evPrim = $c.toByteExact();"
+        } else {
+          s"""
+            if ($c.toByte($wrapper)) {
+              $evPrim = (byte) $wrapper.value;
+            } else {
+              $evNull = true;
+            }
+          """
+        }
         code"""
           UTF8String.IntWrapper $wrapper = new UTF8String.IntWrapper();
```

Comment:
ah we don't need to create int wrapper at all for ansi mode
```
case StringType if ansi => (c, evPrim, evNull) => s"$evPrim = $c.toByteExact();"
case StringType => // the original code
```

Reply:
Done.